File: clean_disney_csv.py
# Disney CSV Robust Cleaner
# Usage: python3 clean_disney_csv.py
import csv

INPUT = 'public/data.csv'
OUTPUT = 'public/data_cleaned.csv'
EXPECTED_COLS = [
    'ID','ITEM','RESTAURANT','CATEGORY','DINING PLAN','LOCATION','DISNEY PARK','DESCRIPTION','PRICE','IS_DDP_SNACK'
]
EXPECTED_LEN = len(EXPECTED_COLS)

def clean_csv(input_path, output_path):
    with open(input_path, newline='', encoding='utf-8') as infile, open(output_path, 'w', newline='', encoding='utf-8') as outfile:
        reader = csv.reader(infile)
        writer = csv.writer(outfile)
        header = next(reader)
        writer.writerow(EXPECTED_COLS)
        for row in reader:
            # Pad or trim row to expected length
            if len(row) < EXPECTED_LEN:
                row += [''] * (EXPECTED_LEN - len(row))
            elif len(row) > EXPECTED_LEN:
                # Try to merge extra columns into DESCRIPTION if possible
                desc_idx = EXPECTED_COLS.index('DESCRIPTION')
                # Merge all extra columns into DESCRIPTION
                row[desc_idx] = ' '.join(row[desc_idx:len(row)-(len(row)-EXPECTED_LEN+1)+1])
                row = row[:EXPECTED_LEN]
            # Clean up boolean and price
            if row[8] in ('true', 'false'):
                row[8] = ''
            if row[9] != 'true':
                row[9] = 'false'
            writer.writerow(row)
    print(f'Cleaned CSV written to {output_path}')

if __name__ == '__main__':
    clean_csv(INPUT, OUTPUT)

File: archive/clean_disney_csv.js
// Disney CSV Cleaner Script
// Run with: node clean_disney_csv.js

const fs = require('fs');
const path = './public/data.csv';
const output = './public/data_cleaned.csv';

const expectedColumns = [
  'ID','ITEM','RESTAURANT','CATEGORY','DINING PLAN','LOCATION','DISNEY PARK','DESCRIPTION','PRICE','IS_DDP_SNACK'
];

function parseCSVLine(line) {
  // Naive CSV parser for quoted fields
  const regex = /"([^"]*)"|([^,]+)/g;
  const result = [];
  let match;
  while ((match = regex.exec(line)) !== null) {
    result.push(match[1] !== undefined ? match[1] : match[2]);
  }
  return result;
}

function cleanCSV(inputPath, outputPath) {
  const lines = fs.readFileSync(inputPath, 'utf8').split(/\r?\n/);
  const header = lines[0];
  const cleaned = [header];
  for (let i = 1; i < lines.length; i++) {
    const line = lines[i];
    if (!line.trim()) continue;
    const fields = parseCSVLine(line);
    if (fields.length !== expectedColumns.length) {
      // Try to fix: pad or trim
      if (fields.length < expectedColumns.length) {
        while (fields.length < expectedColumns.length) fields.push('');
      } else {
        fields.length = expectedColumns.length;
      }
    }
    // Remove 'true'/'false' in price, fix boolean
    if (fields[8] === 'true' || fields[8] === 'false') fields[8] = '';
    if (fields[9] !== 'true') fields[9] = 'false';
    cleaned.push(fields.map(f => f.includes(',') ? '"' + f + '"' : f).join(','));
  }
  fs.writeFileSync(outputPath, cleaned.join('\n'), 'utf8');
  console.log('Cleaned CSV written to', outputPath);
}

cleanCSV(path, output);
